defmt 0.3.7 breaks embedded-hal with the `defmt-03` feature flag

Builds now fail like this:

```rust

error[E0433]: failed to resolve: could not find `defmt` in the list of imported crates
   --> /home/guinea/.cargo/registry/src/index.crates.io-6f17d22bba15001f/embedded-hal-1.0.0/src/i2c.rs:192:41
    |
192 | #[cfg_attr(feature = "defmt-03", derive(defmt::Format))]
    |                                         ^^^^^^^^^^^^^ could not find `defmt` in the list of imported crates
    |
    = note: this error originates in the derive macro `defmt::Format` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider importing this crate through its public re-export
    |
164 + use crate::defmt;
    |

error[E0433]: failed to resolve: could not find `defmt` in the list of imported crates
   --> /home/guinea/.cargo/registry/src/index.crates.io-6f17d22bba15001f/embedded-hal-1.0.0/src/i2c.rs:192:41
    |
192 | #[cfg_attr(feature = "defmt-03", derive(defmt::Format))]
    |                                         ^^^^^^^^^^^^^ could not find `defmt` in the list of imported crates
    |
    = note: this error originates in the derive macro `defmt::Format` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0433]: failed to resolve: could not find `defmt` in the list of imported crates
   --> /home/guinea/.cargo/registry/src/index.crates.io-6f17d22bba15001f/embedded-hal-1.0.0/src/i2c.rs:304:41
    |
304 | #[cfg_attr(feature = "defmt-03", derive(defmt::Format))]
    |                                         ^^^^^^^^^^^^^ could not find `defmt` in the list of imported crates
    |
    = note: this error originates in the derive macro `defmt::Format` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider importing this crate through its public re-export
    |
164 + use crate::defmt;
    |

error[E0433]: failed to resolve: could not find `defmt` in the list of imported crates
   --> /home/guinea/.cargo/registry/src/index.crates.io-6f17d22bba15001f/embedded-hal-1.0.0/src/i2c.rs:304:41
    |
304 | #[cfg_attr(feature = "defmt-03", derive(defmt::Format))]
    |                                         ^^^^^^^^^^^^^ could not find `defmt` in the list of imported crates
    |
    = note: this error originates in the derive macro `defmt::Format` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0433]: failed to resolve: could not find `defmt` in the list of imported crates
   --> /home/guinea/.cargo/registry/src/index.crates.io-6f17d22bba15001f/embedded-hal-1.0.0/src/spi.rs:201:41
    |
201 | #[cfg_attr(feature = "defmt-03", derive(defmt::Format))]
    |                                         ^^^^^^^^^^^^^ could not find `defmt` in the list of imported crates
    |
    = note: this error originates in the derive macro `defmt::Format` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider importing this crate through its public re-export
    |
174 + use crate::defmt;
    |

error[E0433]: failed to resolve: could not find `defmt` in the list of imported crates
   --> /home/guinea/.cargo/registry/src/index.crates.io-6f17d22bba15001f/embedded-hal-1.0.0/src/spi.rs:201:41
    |
201 | #[cfg_attr(feature = "defmt-03", derive(defmt::Format))]
    |                                         ^^^^^^^^^^^^^ could not find `defmt` in the list of imported crates
    |
    = note: this error originates in the derive macro `defmt::Format` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0433]: failed to resolve: could not find `defmt` in the list of imported crates
   --> /home/guinea/.cargo/registry/src/index.crates.io-6f17d22bba15001f/embedded-hal-1.0.0/src/spi.rs:319:41
    |
319 | #[cfg_attr(feature = "defmt-03", derive(defmt::Format))]
    |                                         ^^^^^^^^^^^^^ could not find `defmt` in the list of imported crates
    |
    = note: this error originates in the derive macro `defmt::Format` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider importing this crate through its public re-export
    |
174 + use crate::defmt;
    |

error[E0433]: failed to resolve: could not find `defmt` in the list of imported crates
   --> /home/guinea/.cargo/registry/src/index.crates.io-6f17d22bba15001f/embedded-hal-1.0.0/src/spi.rs:319:41
    |
319 | #[cfg_attr(feature = "defmt-03", derive(defmt::Format))]
    |                                         ^^^^^^^^^^^^^ could not find `defmt` in the list of imported crates
    |
    = note: this error originates in the derive macro `defmt::Format` (in Nightly builds, run with -Z macro-backtrace for more info)
```

`cargo -V` is `cargo 1.80.0-nightly (4de0094ac 2024-05-09)`

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased]
 
+- [#835]: Fix some `defmt` crate name usage
+
 ## [v0.3.7] - 2024-05-13
 
 - [#831]: Fix CI

--- a/macros/src/derives/format/codegen/fields.rs
+++ b/macros/src/derives/format/codegen/fields.rs
@@ -50,18 +50,18 @@ pub(crate) fn codegen(
             .unwrap_or_else(|| format_ident!("arg{}", index));
         // Find the required trait bounds for the field and add the formatting statement depending on the field type and the formatting options
         let bound: Option<syn::Path> = if let Some(FormatOption::Debug2Format) = format_opt {
-            stmts.push(quote!(::defmt::export::fmt(&defmt::Debug2Format(&#ident))));
+            stmts.push(quote!(defmt::export::fmt(&defmt::Debug2Format(&#ident))));
             field_ty.map(|_| parse_quote!(::core::fmt::Debug))
         } else if let Some(FormatOption::Display2Format) = format_opt {
-            stmts.push(quote!(::defmt::export::fmt(&defmt::Display2Format(&#ident))));
+            stmts.push(quote!(defmt::export::fmt(&defmt::Display2Format(&#ident))));
             field_ty.map(|_| parse_quote!(::core::fmt::Display))
         } else if ty == consts::TYPE_FORMAT {
-            stmts.push(quote!(::defmt::export::fmt(#ident)));
-            field_ty.map(|_| parse_quote!(::defmt::Format))
+            stmts.push(quote!(defmt::export::fmt(#ident)));
+            field_ty.map(|_| parse_quote!(defmt::Format))
         } else {
             let method = format_ident!("{}", ty);
-            stmts.push(quote!(::defmt::export::#method(#ident)));
-            field_ty.map(|_| parse_quote!(::defmt::Format))
+            stmts.push(quote!(defmt::export::#method(#ident)));
+            field_ty.map(|_| parse_quote!(defmt::Format))
         };
         if let Some(bound) = bound {
             where_predicates.push(parse_quote!(#field_ty: #bound));

--- a/macros/src/items/timestamp.rs
+++ b/macros/src/items/timestamp.rs
@@ -35,7 +35,7 @@ pub(crate) fn expand(args: TokenStream) -> TokenStream {
         const _: () = {
             #[export_name = "_defmt_timestamp"]
             #[inline(never)]
-            fn defmt_timestamp(fmt: ::defmt::Formatter<'_>) {
+            fn defmt_timestamp(fmt: defmt::Formatter<'_>) {
                 match (#(&(#formatting_exprs)),*) {
                     (#(#patterns),*) => {
                     // NOTE: No format string index, and no finalize call.

